tests/integration/conftest.py

```diff
@@ -29,19 +31,67 @@
 _SDK_ROOT_PATH = Path(__file__).parent.parent.parent.resolve()
 
 
-@pytest.fixture(autouse=True)
-def _reset_and_patch_default_instances() -> None:
-    """Reset the used singletons and patch the default storage client with a temporary directory.
+@pytest.fixture
+def prepare_test_env(monkeypatch: pytest.MonkeyPatch, tmp_path: Path) -> Callable[[], None]:
+    """Prepare the testing environment by resetting the global state before each test.
+
+    This fixture ensures that the global state of the package is reset to a known baseline before each test runs.
+    It also configures a temporary storage directory for test isolation.
+
+    Args:
+        monkeypatch: Test utility provided by pytest for patching.
+        tmp_path: A unique temporary directory path provided by pytest for test isolation.
 
-    To isolate the tests, we need to reset the used singletons before each test case. We also patch the default
-    storage client with a tmp_path.
+    Returns:
+        A callable that prepares the test environment.
     """
-    from crawlee import service_container
 
-    cast(dict, service_container._services).clear()
-    delattr(apify._actor.Actor, '__wrapped__')
+    def _prepare_test_env() -> None:
+        delattr(apify._actor.Actor, '__wrapped__')
+
+        # Set the environment variable for the local storage directory to the temporary path.
+        monkeypatch.setenv(ApifyEnvVars.LOCAL_STORAGE_DIR, str(tmp_path))
+
+        # Reset the flags in the service locator to indicate that no services are explicitly set. This ensures
+        # a clean state, as services might have been set during a previous test and not reset properly.
+        service_locator._configuration_was_retrieved = False
+        service_locator._storage_client_was_retrieved = False
+        service_locator._event_manager_was_retrieved = False
+
+        # Reset the services in the service locator.
+        service_locator._configuration = None
+        service_locator._event_manager = None
+        service_locator._storage_client = None
```

Comment:
If `service_locator` was a `lazy_object_proxy`, we could just re-set `crawlee._service_locator.service_locator` to a new instance of `ServiceLocator` and we wouldn't need to do this here and in crawlee tests (code duplication across different packages is kinda painful)

Reply:
Sure, I have no problem in making a lazy_object_proxy from service_locator if we have a good reason to do so.

Reply:
Cool, here's the issue https://github.com/apify/crawlee-python/issues/884